src/modules/mysql.nix

```diff
@@ -7,28 +7,29 @@ let
   isMariaDB = getName cfg.package == getName pkgs.mariadb;
   format = pkgs.formats.ini { listsAsDuplicateKeys = true; };
   configFile = format.generate "my.cnf" cfg.settings;
-  mysqlOptions = "--defaults-file=${configFile}";
-  mysqldOptions = "${mysqlOptions} --datadir=$MYSQLDATA --basedir=${cfg.package}";
   startScript = pkgs.writeShellScriptBin "start-mysql" ''
     set -euo pipefail
 
-    if [[ ! -d "$MYSQLDATA" ]]; then
-      mkdir -p "$MYSQLDATA"
-      ${if isMariaDB then "${cfg.package}/bin/mysql_install_db" else "${cfg.package}/bin/mysqld"} ${mysqldOptions} ${optionalString (!isMariaDB) "--initialize-insecure"}
+    if [[ ! -d "$MYSQL_HOME" ]]; then
+      mkdir -p "$MYSQL_HOME"
+      ln -sf "${configFile}" "$MYSQL_HOME"/my.cnf
+      ${if isMariaDB then "${cfg.package}/bin/mysql_install_db" else "${cfg.package}/bin/mysqld"} ${optionalString (!isMariaDB) "--initialize-insecure"}
```

Comment:
![image](https://user-images.githubusercontent.com/6224096/204663239-08245fc9-5507-4420-b175-43fecc1af029.png)

datadir is not passed anymore to `mysql_install_db`

Reply:
Hmm, I probably didn't properly clear all of the mysql directory. I've removed the commit that removes the wrapper scripts. I'll keep that for a potentially later PR.